.github/workflows/cpplint.yml

```diff
@@ -0,0 +1,11 @@
+# Borrowed from https://github.com/cpplint/GitHub-Action-for-cpplint?tab=readme-ov-file
+name: cpplint
+on: [push, pull_request]
+jobs:
+  cpplint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+    - run: pip install cpplint
+    - run: cpplint --recursive --exclude='extern/*' --filter=-legal/copyright,-build/header_guard,-runtime/references,-runtime/string,-build/c++11,-runtime/threadsafe_fn,-whitespace/comments,-build/include_subdir,-readability/todo,-readability/casting,-readability/inheritance,-build/include_order,-runtime/int .
```

Comment:
I would suggest defining these settings in a `CPPLINT.cfg` file so that it is possible to run `cpplint` locally using the same settings as the workflow uses.